src/MongoDB.Driver.Core/Core/Connections/IsMasterHelper.cs

```diff
@@ -84,8 +93,8 @@ internal static CommandWireProtocol<BsonDocument> CreateProtocol(BsonDocument is
         {
             try
             {
-                var isMasterResult = await isMasterProtocol.ExecuteAsync(connection, cancellationToken).ConfigureAwait(false);
-                return new IsMasterResult(isMasterResult);
+                var isMasterResultDocument = await isMasterProtocol.ExecuteAsync(connection, cancellationToken).ConfigureAwait(false);
+                return new IsMasterResult(isMasterResultDocument);
             }
```

Comment:
previously we didn't have this catch for isMaster heartbeat calls (there were no catch blocks at all). Can it be a problem? If so, It can be simply fixed :)